src/V3PreLex.l

```diff
@@ -112,6 +116,47 @@ bom             [\357\273\277]
 <INITIAL>"`undef"       { FL_FWDC; return VP_UNDEF; }
 <INITIAL>"`undefineall" { FL_FWDC; return VP_UNDEFINEALL; }
 <INITIAL>"`error"       { FL_FWDC; if (!pedantic()) return VP_ERROR; else return VP_DEFREF; }
+    /* We wanted this to be next to `protect But it must be before `pramga */
+    /* we win only because we both match to the end of the line so the length */
+    /* is equal and we are first*/
+<INITIAL>"`pragma"{wsn}+"protect"[^\n\r]*[\n\r] { FL_FWDC; linenoInc();
+                          if (v3Global.opt.preprocOnly()) {
+                              return VP_TEXT;
+                          }
+                          int res, yypos = 7; /* length of `pragma */
+                          char *yyt = yytext;
+                          yyt[yyleng] = 0; /* this is OK because the last char here is \n or \r */
+                          yyt += yypos;
+                          if ((res = sscanf(yyt, " protect %n", &yypos)) != 0)
+                            yyerrorf("scanf failure");
+                          yyt += yypos;
+                          if (bytes == 0) {
+                            res = sscanf(yyt, "encoding = (enctype = \"BASE64\", line_length = %d, bytes = %d)%n", &length, &bytes, &yypos);
+                            if (res == 2)
+                                return VP_TEXT; /* silently skip it */
+                            bytes = 0;
+                          } else if ((strncmp(yyt, "key_block", 9) == 0) || (strncmp(yyt, "data_block", 10) == 0)) {
+                            BEGIN(BASE64);
+                            return VP_TEXT;
+                          } else {
+                              return VP_TEXT;
+                          }
+                        }
+<BASE64>[A-Za-z0-9+/]+[=]{0,2}[\n\r] {FL_FWDC; linenoInc();
+                        if ((yyleng-1) <= length && ((yyleng & 3) == 1)) {
+                            bytes -= (yyleng-1)/4*3;
+                        } else {
+                            yyerrorf("BASE64 line too long");
+                        }
+                        if (yytext[yyleng-3] == '=')
```

Comment:
I think yyleng can be 2 in some cases so this will break

Reply:
For legal input, `yyleng` minimum length for this rule is actually 5:
If this rule applies, it means we have at least one BASE64 sequence, which is 4 chars + a `\n` or a `\r`.
For illegal input (malformed BASE64) you are right. A single `[A-Za-z0-9+/]` followed by a `[\n\r]` will have a length of 2, and still match the rule. I can fix the rule to use:
```
<BASE64>[A-Za-z0-9+/]{2,}[=]{0,2}[\n\r]
```

Reply:
I think it's legal to have a single base64 character so wouldn't change that.  Just use

      if (yyleng>=3 && yytext[yyleng-3] == '=')

Reply:
As error cases you should perhaps handle a single base64 character. You could use

  if (yyleng>=3 && yytext[yyleng-3] == '=')